src/monattack.cpp

```diff
@@ -697,31 +711,39 @@ void mattack::grow_vine(monster *z, int index)
 
 void mattack::vine(monster *z, int index)
 {
-    if( z->friendly ) {
-        return; // TODO: handle friendly monsters
-    }
     std::vector<point> grow;
     int vine_neighbors = 0;
     z->reset_special(index); // Reset timer
     z->moves -= 100;
     for (int x = z->posx() - 1; x <= z->posx() + 1; x++) {
         for (int y = z->posy() - 1; y <= z->posy() + 1; y++) {
-            if (g->u.posx == x && g->u.posy == y) {
-                if ( g->u.uncanny_dodge() ) {
+            Creature *critter = g->critter_at( x, y );
+            if( critter != nullptr && z->attitude_to( *critter ) != Creature::Attitude::A_FRIENDLY ) {
+                if ( critter->uncanny_dodge() ) {
                     return;
                 } else {
+                    player *foe = dynamic_cast< player* >( critter );
                     body_part bphit = random_body_part();
                     //~ 1$s monster name(vine), 2$s bodypart in accusative
-                    add_msg(m_bad, _("The %1$s lashes your %2$s!"), z->name().c_str(),
-                            body_part_name_accusative(bphit).c_str());
+                    if( critter == &g->u ) {
+                        add_msg( m_bad, _("The %1$s lashes your %2$s!"), z->name().c_str(),
+                                 body_part_name_accusative(bphit).c_str() );
+                    } else if( foe != nullptr ) {
+                        add_msg( _("The %1$s lashes %s's %2$s!"), z->name().c_str(), 
```

Comment:
This block might need either a sees check, or use `add_msg_if_npc`, that function will ignore the message if the creature is not visible to the player. Same for the else branch.